### ISSUE TITLE
FileAlreadyExistsException while updating signatures

### ISSUE THREAD

**Issue by Rainer-73** [dependency-check/DependencyCheck#4426]
Maven goal update-only of dependency-check-maven plugin fails in versions 7.1.0 after downloading signature updates during database maintenance with "java.nio.file.FileAlreadyExistsException: /home/gamma/tmp". (Version 7.0.4 has the same problem, but version 6.5.3 works without problems.)

/home/gamma/tmp is  a symbolic link to the existing temporary directory.
/home/gamma/tmp is set as property java.io.tmpdir.

If java.io.tmpdir points to the link target (the physical directory), then the problem disappeares and update-only goal works as expected. For that reason I think the symbolic link is the root cause of the problem, please make sure symbolic links are evaluated correctly.

environment: RedHat Enterprise Linux 7, Maven 3.8.1, Jdk 1.8

detail information from log:

```
[INFO] Processing Complete for NVD CVE - Modified  (1555 ms)
[INFO] Begin database maintenance
[ERROR] An unexpected SQL Exception occurred; please see the verbose log for more details.
[ERROR] Unexpected SQL Exception
org.owasp.dependencycheck.data.update.exception.UpdateException: Unexpected SQL Exception
    at org.owasp.dependencycheck.data.update.NvdCveUpdater.performUpdate (NvdCveUpdater.java:337)
    at org.owasp.dependencycheck.data.update.NvdCveUpdater.update (NvdCveUpdater.java:136)
    at org.owasp.dependencycheck.Engine.doUpdates (Engine.java:878)
    at org.owasp.dependencycheck.Engine.doUpdates (Engine.java:850)
    at org.owasp.dependencycheck.maven.UpdateMojo.runCheck (UpdateMojo.java:78)
    at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.execute (BaseDependencyCheckMojo.java:995)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.jvnet.hudson.maven3.launcher.Maven35Launcher.main (Maven35Launcher.java:130)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at jenkins.maven3.agent.Maven35Main.launch (Maven35Main.java:178)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:139)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:70)
    at hudson.remoting.UserRequest.perform (UserRequest.java:211)
    at hudson.remoting.UserRequest.perform (UserRequest.java:54)
    at hudson.remoting.Request$2.run (Request.java:376)
    at hudson.remoting.InterceptingExecutorService.lambda$wrap$0 (InterceptingExecutorService.java:78)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
Caused by: org.h2.jdbc.JdbcSQLNonTransientException: IO Exception: "java.nio.file.FileAlreadyExistsException: /home/gamma/tmp"; SQL statement:
UPDATE cpeEntry e SET e.ecosystem=(SELECT cpeEcosystemCache.ecosystem FROM cpeEcosystemCache WHERE cpeEcosystemCache.vendor=e.vendor AND cpeEcosystemCache.product=e.product AND e.ecosystem IS NULL AND cpeEcosystemCache.ecosystem<>'MULTIPLE') WHERE e.ecosystem IS NULL [90028-210]
    at org.h2.message.DbException.getJdbcSQLException (DbException.java:573)
    at org.h2.message.DbException.getJdbcSQLException (DbException.java:496)
    at org.h2.message.DbException.get (DbException.java:216)
    at org.h2.message.DbException.convert (DbException.java:406)
    at org.h2.mvstore.db.MVTempResult.<init> (MVTempResult.java:192)
    at org.h2.mvstore.db.MVPlainTempResult.<init> (MVPlainTempResult.java:69)
    at org.h2.mvstore.db.MVTempResult.of (MVTempResult.java:82)
    at org.h2.result.LocalResult.createExternalResult (LocalResult.java:382)
    at org.h2.result.LocalResult.addRowsToDisk (LocalResult.java:441)
    at org.h2.result.LocalResult.addRowInternal (LocalResult.java:434)
    at org.h2.result.LocalResult.addRowForTable (LocalResult.java:398)
    at org.h2.command.dml.SetClauseList.prepareUpdate (SetClauseList.java:166)
    at org.h2.command.dml.Update.update (Update.java:88)
    at org.h2.command.dml.DataChangeStatement.update (DataChangeStatement.java:61)
    at org.h2.command.CommandContainer.update (CommandContainer.java:174)
    at org.h2.command.Command.executeUpdate (Command.java:252)
    at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal (JdbcPreparedStatement.java:209)
    at org.h2.jdbc.JdbcPreparedStatement.executeUpdate (JdbcPreparedStatement.java:169)
    at org.apache.commons.dbcp2.DelegatingPreparedStatement.executeUpdate (DelegatingPreparedStatement.java:136)
    at org.apache.commons.dbcp2.DelegatingPreparedStatement.executeUpdate (DelegatingPreparedStatement.java:136)
    at org.owasp.dependencycheck.data.nvdcve.CveDB.cleanupDatabase (CveDB.java:1265)
    at org.owasp.dependencycheck.data.update.NvdCveUpdater.performUpdate (NvdCveUpdater.java:335)
    at org.owasp.dependencycheck.data.update.NvdCveUpdater.update (NvdCveUpdater.java:136)
    at org.owasp.dependencycheck.Engine.doUpdates (Engine.java:878)
    at org.owasp.dependencycheck.Engine.doUpdates (Engine.java:850)
    at org.owasp.dependencycheck.maven.UpdateMojo.runCheck (UpdateMojo.java:78)
    at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.execute (BaseDependencyCheckMojo.java:995)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.jvnet.hudson.maven3.launcher.Maven35Launcher.main (Maven35Launcher.java:130)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at jenkins.maven3.agent.Maven35Main.launch (Maven35Main.java:178)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:139)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:70)
    at hudson.remoting.UserRequest.perform (UserRequest.java:211)
    at hudson.remoting.UserRequest.perform (UserRequest.java:54)
    at hudson.remoting.Request$2.run (Request.java:376)
    at hudson.remoting.InterceptingExecutorService.lambda$wrap$0 (InterceptingExecutorService.java:78)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)
Caused by: java.nio.file.FileAlreadyExistsException: /home/gamma/tmp
    at sun.nio.fs.UnixException.translateToIOException (UnixException.java:88)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:102)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:107)
    at sun.nio.fs.UnixFileSystemProvider.createDirectory (UnixFileSystemProvider.java:384)
    at java.nio.file.Files.createDirectory (Files.java:674)
    at java.nio.file.Files.createAndCheckIsDirectory (Files.java:781)
    at java.nio.file.Files.createDirectories (Files.java:727)
    at org.h2.store.fs.disk.FilePathDisk.createTempFile (FilePathDisk.java:435)
    at org.h2.store.fs.FileUtils.createTempFile (FileUtils.java:404)
    at org.h2.mvstore.db.MVTempResult.<init> (MVTempResult.java:178)
    at org.h2.mvstore.db.MVPlainTempResult.<init> (MVPlainTempResult.java:69)
    at org.h2.mvstore.db.MVTempResult.of (MVTempResult.java:82)
    at org.h2.result.LocalResult.createExternalResult (LocalResult.java:382)
    at org.h2.result.LocalResult.addRowsToDisk (LocalResult.java:441)
    at org.h2.result.LocalResult.addRowInternal (LocalResult.java:434)
    at org.h2.result.LocalResult.addRowForTable (LocalResult.java:398)
    at org.h2.command.dml.SetClauseList.prepareUpdate (SetClauseList.java:166)
    at org.h2.command.dml.Update.update (Update.java:88)
    at org.h2.command.dml.DataChangeStatement.update (DataChangeStatement.java:61)
    at org.h2.command.CommandContainer.update (CommandContainer.java:174)
    at org.h2.command.Command.executeUpdate (Command.java:252)
    at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal (JdbcPreparedStatement.java:209)
    at org.h2.jdbc.JdbcPreparedStatement.executeUpdate (JdbcPreparedStatement.java:169)
    at org.apache.commons.dbcp2.DelegatingPreparedStatement.executeUpdate (DelegatingPreparedStatement.java:136)
    at org.apache.commons.dbcp2.DelegatingPreparedStatement.executeUpdate (DelegatingPreparedStatement.java:136)
    at org.owasp.dependencycheck.data.nvdcve.CveDB.cleanupDatabase (CveDB.java:1265)
    at org.owasp.dependencycheck.data.update.NvdCveUpdater.performUpdate (NvdCveUpdater.java:335)
    at org.owasp.dependencycheck.data.update.NvdCveUpdater.update (NvdCveUpdater.java:136)
    at org.owasp.dependencycheck.Engine.doUpdates (Engine.java:878)
    at org.owasp.dependencycheck.Engine.doUpdates (Engine.java:850)
    at org.owasp.dependencycheck.maven.UpdateMojo.runCheck (UpdateMojo.java:78)
    at org.owasp.dependencycheck.maven.BaseDependencyCheckMojo.execute (BaseDependencyCheckMojo.java:995)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.jvnet.hudson.maven3.launcher.Maven35Launcher.main (Maven35Launcher.java:130)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at jenkins.maven3.agent.Maven35Main.launch (Maven35Main.java:178)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:139)
    at hudson.maven.Maven3Builder.call (Maven3Builder.java:70)
    at hudson.remoting.UserRequest.perform (UserRequest.java:211)
    at hudson.remoting.UserRequest.perform (UserRequest.java:54)
    at hudson.remoting.Request$2.run (Request.java:376)
    at hudson.remoting.InterceptingExecutorService.lambda$wrap$0 (InterceptingExecutorService.java:78)
    at java.util.concurrent.FutureTask.run (FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
    at java.lang.Thread.run (Thread.java:748)

```

**Comment by aikebah** [comment]
The root cause is resolved upstream in release H2 v2.1.214

**Comment by Rainer-73** [comment]
It took a long time but the dependency-check-maven plugin version 7.1.2 is available in mavencentral now. I can confirm the problem is solved. Thank you very much.
